schema/base-model.js

```diff
@@ -66,9 +66,9 @@ class BaseModel extends Model {
       .whereNot(`${this.tableName}.deleted`, null);
   }
 
-  static count(establishmentId, query) {
+  static count(value, query, key = 'establishmentId') {
     return (query || this.query())
-      .where({ establishmentId })
+      .where({ [key]: value })
```

Comment:
This feels very dangerous. I'm not sure changing the base model is the right approach here given that I'm guessing it's only the `Procedure` model that needs this.

We should extend the `count` function in the models that need a custom implementation rather than changing all the models and having to also go and change all the implementations (although I imagine you've already done some of that)

Reply:
this does actually default to the previous behaviour, but you are right, this was a bit silly, I'll kill it

Reply:
Oh right, I see now.

I'm less bothered then. I was misreading the code.